File: app/src/main/java/com/falcon/restaurants/screens/common/dialogs/ServerErrorDialogFragment.java
package com.falcon.restaurants.screens.common.dialogs;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import androidx.appcompat.app.AlertDialog;

public class ServerErrorDialogFragment extends BaseDialog {

    static final String ERROR = "ERROR";

    public ServerErrorDialogFragment() {}

    public static ServerErrorDialogFragment newInstance(String error) {
        ServerErrorDialogFragment fragment = new ServerErrorDialogFragment();
        Bundle args = new Bundle();
        args.putString(ERROR, error);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        String error = getArguments().getString(ERROR);

        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(getActivity());
        alertDialogBuilder.setTitle("Server Error");
        alertDialogBuilder.setMessage("Error: " + error);
        alertDialogBuilder.setPositiveButton("OK",  new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                // on success
            }
        });

        alertDialogBuilder.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {

            @Override
            public void onClick(DialogInterface dialog, int which) {
                if (dialog != null) {
                    dialog.dismiss();
                }
            }
        });
        return alertDialogBuilder.create();
    }

}

File: app/src/main/java/com/falcon/restaurants/screens/common/dialogs/ServerErrorDialogFragment.kt
package com.falcon.restaurants.screens.common.dialogs
import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog

class ServerErrorDialogFragment : BaseDialog() {

    companion object{
        const val ERROR: String = "ERROR"
        fun newInstance(error: String): ServerErrorDialogFragment {
            val fragment = ServerErrorDialogFragment()
            val args = Bundle()
            args.putString(ERROR, error)
            fragment.arguments =args
            return fragment
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        val error: String  = arguments!!.getString(ERROR) as String

        val alertDialogBuilder: AlertDialog.Builder = AlertDialog.Builder(requireActivity())
        alertDialogBuilder.setTitle("Server Error")
        alertDialogBuilder.setMessage("Error: $error")

        alertDialogBuilder.setPositiveButton("OK"
        ) { dialog, which -> TODO("Not yet implemented") }

        alertDialogBuilder.setNegativeButton("Cancel"
        ) { dialog, which ->
            if (dialog != null) {
                dialog.dismiss()
            }
        }

        return alertDialogBuilder.create()
    }

}